packages/snaps-sdk/src/internals/errors.ts

```diff
@@ -74,15 +125,9 @@ export function getErrorCode(error: unknown) {
  * @internal
  */
 export function getErrorData(error: unknown) {
-  if (
-    isObject(error) &&
-    hasProperty(error, 'data') &&
-    typeof error.data === 'object' &&
-    error.data !== null &&
-    isValidJson(error.data) &&
-    !Array.isArray(error.data)
-  ) {
-    return error.data;
+  const value = getObjectProperty(error, 'data');
+  if (value !== null && isValidJson(value) && !Array.isArray(value)) {
```

Comment:
Do we actually need to check for full JSON validity here?

Reply:
It's what we did previously, I'm not sure if it's _really_ needed, but kept it to be on the safe side.